core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -446,20 +446,16 @@ private void commitSimpleTransaction() {
       }
 
       Set<String> committedFiles = committedFiles(ops, newSnapshots);
-      if (committedFiles != null) {
-        // delete all of the files that were deleted in the most recent set of operation commits
-        Tasks.foreach(deletedFiles)
-            .suppressFailureWhenFinished()
-            .onFailure((file, exc) -> LOG.warn("Failed to delete uncommitted file: {}", file, exc))
-            .run(
-                path -> {
-                  if (!committedFiles.contains(path)) {
-                    ops.io().deleteFile(path);
-                  }
-                });
-      } else {
-        LOG.warn("Failed to load metadata for a committed snapshot, skipping clean-up");
-      }
+      // delete all of the files that were deleted in the most recent set of operation commits
+      Tasks.foreach(deletedFiles)
+          .suppressFailureWhenFinished()
+          .onFailure((file, exc) -> LOG.warn("Failed to delete uncommitted file: {}", file, exc))
+          .run(
+              path -> {
+                if (committedFiles == null || !committedFiles.contains(path)) {
```

Comment:
I just realized alternatively we could've just returned an empty set in `committedFiles` instead of null and then could've removed the  (committedFiles == null) check (in addition to the current top level check). I'm not super opinionated on that though (it can be done in a follow on)

Reply:
I think this should actually change `committedFiles` to return `ImmutableSet.of()` if there are no new snapshot IDs. The logic is correct to warn if the other reason null is returned happens (a committed snapshot is missing). `null` signals that the output of the method is invalid, which we assumed was the case if there are no committed snapshots. But here we have a case where it's a valid case to have no committed snapshots and therefore no committed files.

Reply:
I have a PR https://github.com/apache/iceberg/pull/9221 for addressing this.